Source/WebInspectorUI/UserInterface/Views/OverrideDeviceSettingsPopover.js

```diff
@@ -0,0 +1,321 @@
+            ],
+        };
+
+        const compatibilitySettingsGroup = {
+            name: WI.UIString("Compatibility"),
+            settings: [
+                {name: WI.UIString("Disable site-specific hacks"), setting: InspectorBackend.Enum.Page.Setting.NeedsSiteSpecificQuirks, value: false},
+            ],
+        };
+
+        const securitySettingsGroup = {
+            name: WI.UIString("Security"),
+            settings: [
+                {name: WI.UIString("Disable cross-origin restrictions"), setting: InspectorBackend.Enum.Page.Setting.WebSecurityEnabled, value: false},
+            ],
+        };
+
+        const privacySettingsGroup = {
+            name: WI.UIString("Privacy"),
+            settings: [
+                {name: WI.UIString("Enable Intelligent Tracking Prevention debug mode"), setting: InspectorBackend.Enum.Page.Setting.ITPDebugModeEnabled, value: true},
+            ],
+        };
+
+        // COMPATIBILITY (iOS 14.0): `Page.Setting.AdClickAttributionDebugModeEnabled` was renamed to `Page.Setting.PrivateClickMeasurementDebugModeEnabled`.
+        if (InspectorBackend.Enum.Page.Setting.PrivateClickMeasurementDebugModeEnabled)
+            privacySettingsGroup.settings.push({name: WI.UIString("Enable Private Click Measurement debug mode"), setting: InspectorBackend.Enum.Page.Setting.PrivateClickMeasurementDebugModeEnabled, value: true});
+
+        // COMPATIBILITY (iOS 14.0): `Page.Setting.AdClickAttributionDebugModeEnabled` was renamed to `Page.Setting.PrivateClickMeasurementDebugModeEnabled`.
+        if (InspectorBackend.Enum.Page.Setting.AdClickAttributionDebugModeEnabled)
+            privacySettingsGroup.settings.push({name: WI.UIString("Ad Click Attribution debug mode"), setting: InspectorBackend.Enum.Page.Setting.AdClickAttributionDebugModeEnabled, value: true});
+
+        const mediaSettingsGroup = {
+            name: WI.UIString("Media"),
+            settings: [
+                {name: WI.UIString("Allow media capture on insecure sites"), setting: InspectorBackend.Enum.Page.Setting.MediaCaptureRequiresSecureConnection, value: false},
+                {name: WI.UIString("Disable ICE candidate restrictions"), setting: InspectorBackend.Enum.Page.Setting.ICECandidateFilteringEnabled, value: false},
+                {name: WI.UIString("Use mock capture devices"), setting: InspectorBackend.Enum.Page.Setting.MockCaptureDevicesEnabled, value: true},
+                {name: WI.UIString("Disable encryption"), setting: InspectorBackend.Enum.Page.Setting.WebRTCEncryptionEnabled, value: false},
+            ],
+        };
+
+        let settingsGroups = [coreSettingsGroup];
+
+        if (InspectorFrontendHost.isRemote) {
+            settingsGroups.push(compatibilitySettingsGroup);
+            settingsGroups.push(securitySettingsGroup);
+            settingsGroups.push(privacySettingsGroup);
+        }
```

Comment:
I wonder if we should instead create `settingsGroups` at the start and `push` the various groups directly into it conditionally based on `isRemote` to avoid doing the work to create the unused groups for non-remote cases (which is most of the time)? This would also make it clearer at the time the group is declared if it is present for remote inspection or not.

Reply:
Done